src/layers/sds/domain/base.py

```diff
@@ -67,6 +73,50 @@ def _validate_object_class(cls: "SdsBaseModel", values: dict) -> dict:
     return values
 
 
+def _get_field_name_for_alias(cls: BaseModel, alias):
+    try:
+        (field_name,) = (
+            field_name
+            for field_name, model_field in cls.__fields__.items()
+            if model_field.alias == alias
+        )
+    except ValueError:
+        raise AliasLookupError(f"No field with alias '{alias}' found")
+    return field_name
+
+
+def _get_alias_for_field_name(cls: BaseModel, field):
```

Comment:
Just a question for my understanding - what's the purpose of the alias'? And like what scenario would they be used in?

Reply:
in the models, the "aliases" are the names coming from the ldap server (e.g. `nhsasclient`) whereas the field names are a bit more readable (`nhs_as_client`). When doing `add` operations we pass the whole lot in for pydantic to sort out (so `nhsasclient` gets converted to `nhs_as_client`) but for modify we're given `nhsasclient` and then we need to manually (via the model) to convert to `nhs_as_client`